hudi-common/src/main/java/org/apache/hudi/common/model/AWSDmsAvroPayload.java

```diff
@@ -69,7 +75,15 @@ private Option<IndexedRecord> handleDeleteOperation(IndexedRecord insertValue) t
 
   @Override
   public Option<IndexedRecord> getInsertValue(Schema schema, Properties properties) throws IOException {
-    return getInsertValue(schema);
+    boolean dropPartCol = Boolean.parseBoolean(properties.getProperty(HoodieTableConfig.DROP_PARTITION_COLUMNS.key()));
```

Comment:
This is not the right way to fix this: 

- Record payload itself should not be manipulating the schema -- it should rely on the correctness of the schema passed in externally. 
- It should be responsibility of the caller it uses appropriate schema to decode the record